src/components/ControlledInput/ControlledInput.ts

```diff
@@ -0,0 +1,64 @@
+import { Block } from "../../core";
+
+import { validateValue } from "../../utils/validator";
+
+import Error from "../Error";
+
+import { inputTemplate } from "./ControlledInput.tmpl";
+import { ControlledInputProps } from "./ControlledInput.types";
+
+export default class ControlledInput extends Block {
+  static componentName = "ControlledInput";
+
+  public isValid = true;
+
+  constructor({ validationRule, ...props }: ControlledInputProps) {
+    super({
+      ...props,
+      onFocus: (evt: FocusEvent) => {
+        const { value } = evt.target as HTMLInputElement;
+        const errorElement = this.getErrorElement();
+
+        if (validationRule && errorElement) {
+          const errorText = validateValue(validationRule, value);
+          if (errorText !== "") {
+            this.isValid = false;
+          } else {
+            this.isValid = true;
+          }
+          errorElement.setProps({ errorText });
+        }
+      },
+      onBlur: (evt: FocusEvent) => {
+        const { value } = evt.target as HTMLInputElement;
+        const errorElement = this.getErrorElement();
+
+        if (validationRule && errorElement) {
+          const errorText = validateValue(validationRule, value);
+          if (errorText !== "") {
+            this.isValid = false;
+          } else {
+            this.isValid = true;
+          }
```

Comment:
Можно сократить
```suggestion
          this.isValid = errorText === ""
```
Вообще лучше вынести это в отдельную функцию, потому что эта логика часто используется